cmd/dogstatsd/subcommands/start/command_test.go

```diff
@@ -5,21 +5,25 @@
 
 package start
 
-import (
-	"testing"
+// TODO: (components) This test is does not work with the current error handling
```

Comment:
While https://github.com/DataDog/datadog-agent/pull/15331 improves the error message UX - I don't think it will fix the problem with this test. This test will fail because `start` is never called because `config` returns an error before dependencies are fulfilled.

Reply:
Maybe https://github.com/DataDog/datadog-agent/pull/15346 helps?

Reply:
I think that may work. Ill update this PR once yours merges and take the change.